app/components/blacklight/document/page_header_component.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module Blacklight
+  module Document
+    # Render the start over and prev/next displays
+    class PageHeaderComponent < Blacklight::Component
+      attr_reader :document, :blacklight_config, :search_context, :search_session
+
+      def initialize(document:, search_context:, search_session:)
+        super
+        @search_context = search_context
+        @search_session = search_session
+        @document = document
+      end
+
+      def render?
+        search_context.present? || search_session.present? || has_header_tools?
+      end
+
+      def before_render
+        @blacklight_config = helpers.blacklight_config
+      end
+
+      def applied_params_component
+        return unless blacklight_config.track_search_session.applied_params_component
+
+        blacklight_config.track_search_session.applied_params_component.new
+      end
+
+      def pagination_component
+        return unless blacklight_config.track_search_session.item_pagination_component
+
+        blacklight_config.track_search_session.item_pagination_component.new(search_context: search_context, search_session: search_session, current_document: document)
+      end
+
+      def has_header_tools?
+        header_actions.any? || show_header_tools_component
+      end
+
+      def pagination_container_classes
+        has_header_tools? ? 'col-12 col-md-6 ms-auto' : ''
+      end
+
+      def header_container_classes
+        has_header_tools? ? 'd-flex justify-content-between pagination-search-widgets pb-2 row justify-content-end' : 'pagination-search-widgets'
+      end
+
+      def header_actions
+        actions = helpers.send(:filter_partials, blacklight_config.view_config(:show).header_actions, { document: document })
+        actions.map { |_k, v| v }
+      end
+
+      def show_header_tools_component
+        blacklight_config.view_config(:show).show_header_tools_component
+      end
+
+      def render_header_tools
+        return unless has_header_tools?
+
+        show_header_tools_component&.tap do |show_tools_component_class|
+          return render show_tools_component_class.new(document: document)
+        end
+
+        render Blacklight::Document::ActionsComponent.new(document: document,
+                                                          tag: 'div',
+                                                          classes: 'd-inline-flex header-tools align-items-center col-12 col-md-6 ms-auto justify-content-md-end',
```

Comment:
Extracting these classes (as well as the component class itself) to a separate method makes it easier for people to configure via overriding.

Reply:
Done.